financial-templates-lib/logger/PagerDutyTransport.js

```diff
@@ -0,0 +1,33 @@
+// This transport enables winston logging to send messages to pager duty.
+
+const Transport = require("winston-transport");
+const pdClient = require("node-pagerduty");
+
+module.exports = class PagerDutyTransport extends Transport {
+  constructor(winstonOpts, pagerDutyOptions) {
+    super(winstonOpts);
+    this.serviceId = pagerDutyOptions.pdServiceId;
+    this.fromEmail = pagerDutyOptions.fromEmail;
+    this.pd = new pdClient(pagerDutyOptions.pdApiToken);
+  }
+
+  async log(info, callback) {
+    // TODO: refactor this processing to better parse mrkdwn and complex data structure from winston.
+    await this.pd.incidents.createIncident(this.fromEmail, {
```

Comment:
Just checking, is this consistent with how you handle errors in the `log()` methods for other transports?

Reply:
Ye so the `log` function gets called whenever the logging event occurs from Winston. This basically means that after the logging is initiated the pager duty library is called.